src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -98,19 +114,26 @@ private void checkHealth() {
     }
   }
 
-  public void stop() {
+  public synchronized void stop() {
```

Comment:
I am not sure we need to check `_isShutdown ` in `stop()` and `awaitShutdown()`. Can you explain the reason of this check?